### PR TITLE
refactor(framework): add scopeHelper

### PR DESCRIPTION
**Pull request by warren830**
### Summary
This pr mainly work on refactor github scope api, and extract a scopeHelper which can be used in other plugins

### Does this close any open issues?
Part of #4653

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Review by likyh — APPROVED** [review]
Let's have a try.

File: backend/helpers/pluginhelper/api/scope_helper.go [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+	}
+
+	// Save the scopes to the database
+	return c.save(scopesValue.Interface(), c.db.Create)
+}
+
+func (c *ScopeApiHelper) Update(input *plugin.ApiResourceInput, fieldName string, connection interface{}, scope interface{}) errors.Error {
+	connectionId, scopeId := ExtractParam(input.Params)
+
+	if connectionId == 0 || len(scopeId) == 0 || scopeId == "0" {
```
Comment (mindlesscloud) [inline backend/helpers/pluginhelper/api/scope_helper.go:127]: Shall we check for `scopeId == ""`？

**Review by mindlesscloud — APPROVED** [review]
LGTM

File: backend/helpers/pluginhelper/api/scope_helper.go [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+	connectionId, err := strconv.ParseUint(params["connectionId"], 10, 64)
+	if err != nil {
+		return 0, ""
+	}
+	scopeId := params["scopeId"]
+	return connectionId, scopeId
+}
+
+// VerifyPrimaryKeyValue function verifies that the primary key value of a given struct instance is not zero or empty.
+func VerifyPrimaryKeyValue(i interface{}) errors.Error {
```
Comment (klesh) [inline backend/helpers/pluginhelper/api/scope_helper.go:267]: I don't think it is working correctly, the primary keys are most likely nested inside fields, consider using instead https://github.com/apache/incubator-devlake/blob/243cc8a80aa5b37828e2a142ac9f7e3269b7e1dc/backend/core/utils/structfield.go#L25

File: backend/helpers/pluginhelper/api/scope_helper.go [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+	if err != nil {
+		if c.db.IsDuplicationError(err) {
+			return errors.BadInput.New("the scope already exists")
+		}
+		return err
+	}
+	return nil
+}
+
+func ExtractParam(params map[string]string) (uint64, string) {
```
Comment (klesh) [inline backend/helpers/pluginhelper/api/scope_helper.go:257]: This is specifically for extracting fields for `scopeHelper`
It should NOT be a public method and with this very vague naming.

**Pull request merged**